src/qt/sendcoinsdialog.cpp

```diff
@@ -318,11 +318,24 @@ void SendCoinsDialog::on_sendButton_clicked()
         // append fee string if a fee is required
         questionString.append("<hr /><span style='color:#aa0000;'>");
         questionString.append(BitcoinUnits::formatHtmlWithUnit(model->getOptionsModel()->getDisplayUnit(), txFee));
+
         questionString.append("</span> ");
         questionString.append(tr("added as transaction fee"));
 
+        double txSize = (double)currentTransaction.getTransactionSize();
+
+        // safe to assume that txSize cannot be 0 at this point?
+        double satPerVByte = txFee / txSize; // txFee is already in satoshis
```

Comment:
From the isolated code part perspective, I'd say either factor out into a function that handles the txSize validity or so an if structure above to ensue txSize is valid.

Reply:
Moved sat/vbyte calculation to the WalletModelTransaction class in e8485d10e63c1c66ea85b253ff38ee86fd2a3ea6